src/parser/core/Combatant.ts

```diff
@@ -71,10 +71,15 @@ class Combatant extends Entity {
     return SPECS[this.specId];
   }
   get race(): Race | null {
-    if (!this.owner.characterProfile) {
+    if (!this.owner.characterProfile || !this.owner.characterProfile.race) {
       return null;
     }
-    const raceId = this.owner.characterProfile.race;
+    let raceId = this.owner.characterProfile.race;
+    if (raceId === null) {
```

Comment:
Thanks for implementing the previous suggestion. I reckon that should make this code (and the "anonymous" race) obsolete. Can it be removed without affecting the behavior?

Reply:
This causes issues later on within the code where race ID is expected to be set to something other than null/undefined. That is why I opted for an Anonymous race.

Reply:
That is strange. `null` should be a common value for race as we often can't fetch the character profile, so it should work just fine. Do you have an example of something that would break?

Reply:
Look at the lines just below this, it breaks and from what I experienced it was just a ripple affect.